mbedtls/src/wrapper_macros.rs

```diff
@@ -109,6 +113,32 @@ macro_rules! define_enum {
 }
 
 macro_rules! define_struct {
+    { define_custom $(#[$m:meta])* struct $name:ident $(lifetime $l:tt)* inner $inner:ident members $($(#[$mm:meta])* $member:ident: $member_type:ty,)* } => {
+        as_item!(
+        #[allow(dead_code)]
+        $(#[$m])*
+        pub struct $name<$($l)*> {
+            $($(#[$mm])* $member: $member_type,)*
+        }
+        );
+
+        as_item!(
+        #[allow(dead_code)]
+        impl<$($l)*> $name<$($l)*> {
+            pub(crate) fn handle(&self) -> &::mbedtls_sys::$inner {
+                self.inner.handle()
```

Comment:
It's not appropriate to use `inner` here since that field is defined outside of the macro.

I don't think this macro adds much for the case where we're wrapping another type that's already using the macro. Just implement it by hand. No need to `impl Send` I think as it's automatically derived.

Reply:
Handle is useful to integrate with other common macros from same file such as below:

```
    { into $name:ident inner $inner:ident $(lifetime $l:tt)* lifetime2 $l2:tt } => {
        as_item!(
        impl<$l2,$($l),*> Into<*const $inner> for &$l2 $name<$($l)*> {
            fn into(self) -> *const $inner {
                self.handle()
            }
        }
        );
```

Reply:
So implement handle manually.